contracts/lifecycle/TimeFrame.sol

```diff
@@ -0,0 +1,70 @@
+pragma solidity ^0.5.0;
+
+/**
+ * @title Provides generic functionality for a TimeFrame.
+ * @author Sam Ruberti <sam@GRAFFT.io>
+ * @dev This library provides useful methods for governing a start and an end time.
+ *
+ * WARNING: This should NOT be used for contracts that depend on time for an outcome where a miner can benefit.
+ * One example would be a lottery where the first submission of an answer to a problem is rewarded. In this case the
+ * contract would be vulnerable to malicious miners who could read the answer of a not-yet-mined transaction then
+ * substitute their own answer in a new transaction before the victim's transaction, stealing the reward in the process.
+ * The miner could steal the reward.
+ *
+ * SAFE USE: TimeFrame CAN be used for approximate times that govern periods when methods and actions may occur.
+ * An example is an escrow that contains a start and expiration date. You can read about timestamp security
+ * vulnerabilities here:
+ * github.com/ethereumbook/ethereumbook/blob/develop/09smart-contracts-security.asciidoc#block-timestamp-manipulation
```

Comment:
```suggestion
  * https://github.com/ethereumbook/ethereumbook/blob/first_edition_first_print/09smart-contracts-security.asciidoc#block-timestamp-manipulation
```

I changed it so that it points to the tag of the first edition of the book as opposed to `develop`, which may change in the future.

Reply:
replaced it with your suggestion